src/steps/pre-query/parse-request-primary.js

```diff
@@ -22,14 +22,24 @@ export default function(data, parseAsLinkage) {
     }
 
     catch (error) {
-      const title = "The resources you provided could not be parsed.";
-      const details = `The precise error was: "${error.message}".`;
-      reject(new APIError(400, undefined, title, details));
+      if (error instanceof APIError) {
+        reject(error);
+      }
+
+      else {
+        const title = "The resources you provided could not be parsed.";
+        const details = `The precise error was: "${error.message}".`;
+        reject(new APIError(400, undefined, title, details));
+      }
     }
   });
 }
 
-function relationshipObjectFromJSON(json) {
+function relationshipObjectFromJSON(json, relationship) {
+  if (typeof json.data === "undefined") {
+    throw new APIError(400, undefined, `No data key was found for the '${relationship}' relationship.`);
```

Comment:
Sure, although I have tried writing integration tests for this library a few times now but something about the way it's set up means I keep getting silent errors or a failures of the first test without any pointers to what I'm doing wrong.

The main thing that you're doing that's different from what I'm used to is structuring your tests like this

```javascript
describe('' => {
  makeRequest().then((res) => {
    describe('Stuff', () => {
      it('should do something', () => {
        expect(true).to.be.ok;
      });
    });
  });
});
```

not like this

```javascript
describe('Stuff', () => {
  var res;

  before((done) => {
    makeRequest('/users', (err, result) => {
      res = result;
      done(err);
    });
  });

  it('should do something', () => {
    expect(res).to.be.ok;
  });
});
```

The main problem with the first method is that if `makeRequest()` rejects then the tests aren't run. The test suite exits saying slightly fewer than normal tests have passed, but none have failed.

I might be misunderstanding something here, but this was was the main problem I was having when trying to write tests for #70. Are you able to explain some of the reasoning for that structure?